lxd/device/proxy.go

```diff
@@ -181,13 +182,51 @@ func (d *proxy) validateEnvironment() error {
 	return nil
 }
 
+// validateNetworkForwardConflicts checks that the proxy device about to be created does not
+// overlap on existing network forward (both entities can't have the same listening address with
+// the same port number)
+func (d *proxy) validateNetworkForwardConflicts() error {
```

Comment:
I would call this something more generic like `validateListenAddressConflicts()` and then we can expand it later to cover other things.